### PR TITLE
TTVarGlyphset

### PR DESCRIPTION
**Pull request by behdad**
Just, this is pretty much your code integrated into TTFont. Needs some clean up still, but I think I like the general structure. I did minor testing. Can you do some more?

### CODE REVIEW

**Comment by behdad** [comment]
I used this in the `varLib.interpolatable` tool already, seems to be working.

**Comment by behdad** [comment]
I removed the compile though, used deepcopy instead.

**Comment by behdad** [comment]
> I removed the compile though, used deepcopy instead.

And now a shallow copy.

**Comment by justvanrossum** [comment]
Thanks for picking this up!

I'm not immediately able to contribute more testing. We probably have some test fonts in the the test suite that can be used for this?

**Comment by behdad** [comment]
This is ready now. I visually tested it with the svgPathPen module as well.

**Comment by justvanrossum** [comment]
Looks good to me, but I would prefer more test coverage before we merge.

**Comment by behdad** [comment]
Sure. I'll add a subset RobotoFlex font with just letter 'I' and test it at various locations.

**Comment by behdad** [comment]
Tests added.

**Review by justvanrossum — COMMENTED** [review]
LGTM, would wait for review by Cosimo, though.

File: Lib/fontTools/ttLib/ttFont.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttFont.py:682]: docstring could mention the new location and normalized parameters, and the fact that currently this only works for TrueType-flavored variable fonts. Maybe you could add some TODO or file an issue for CFF2 support

File: Lib/fontTools/ttLib/ttGlyphSet.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttGlyphSet.py:115]: would be nice of `avar` had a `mapLocation` or something that did the same so we can share this logic in few other places (not necessarily in this same PR)

File: Lib/fontTools/ttLib/ttGlyphSet.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttGlyphSet.py:208]: could be moved to the top-level scope

File: Lib/fontTools/ttLib/ttGlyphSet.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttGlyphSet.py:144]: looks almost identical to `glyf._setCoordinates`, ideally we'd factor out the shared code. Maybe not worth in this case. I see you don't round here, whereas glyf._setCoordinates does, but I guess that's intentional.

File: Lib/fontTools/ttLib/ttGlyphSet.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttGlyphSet.py:182]: maybe initialize these attributes to None or 0, and comment that the get updated once you draw the glyph.

File: Lib/fontTools/varLib/interpolatable.py
Comment (anthrotype) [inline Lib/fontTools/varLib/interpolatable.py:396]: there's one parenthesis too much in **(** sorted(...) **)**

**Review by anthrotype — APPROVED** [review]
LGTM, with some minor comments

File: Lib/fontTools/ttLib/ttGlyphSet.py
Comment (behdad) [reply Lib/fontTools/ttLib/ttGlyphSet.py:144]: I can add rounding just to be similar to the other case. It doesn't need rounding. I copied code from Just.

You are right that ideally this code should be refactored.

File: Lib/fontTools/ttLib/ttFont.py
Comment (anthrotype) [inline Lib/fontTools/ttLib/ttFont.py:712]: I think if `location is not None and "gvar" not in self` (i.e. non variable font) we should error out instead of silently return the default glyph set

Comment (behdad) [reply Lib/fontTools/ttLib/ttFont.py:712]: > I think if `location is not None and "gvar" not in self` (i.e. non variable font) we should error out instead of silently return the default glyph set

That's debatable.  What if location points to an axis that doesn't exist in the varfont then? That's the same as location and no varfont... So I'm leaning towards the code as is.

**Pull request merged**
